src/xdp/trn_agent_xdp.c

```diff
@@ -260,6 +266,16 @@ static __inline int trn_encapsulate(struct transit_packet *pkt,
 	__builtin_memset(&pkt->scaled_ep_opt->scaled_ep_data, 0,
 			 sizeof(struct trn_gnv_scaled_ep_data));
 
+	pkt->pod_label_value_opt->opt_class = TRN_GNV_OPT_CLASS;
+	pkt->pod_label_value_opt->type = TRN_GNV_LABEL_VALUE_OPT_TYPE;
```

Comment:
How do we distinguish between pod label and namespace label?

Reply:
There is option named pod_label_value_opt and namespace_label_value_opt which is to distinguish them. For the object they share same structure as designed.

Reply:
One the ingress side, how do you know whether you are looking at pod label or namespace label? They both appear to use TRN_GNV_LABEL_VALUE_OPT_TYPE.

Reply:
In this pr there is code showing how to retrieve from ingress side. Since there are two separate options, the distinguish is point location.

Reply:
We tried to optimize this by not adding Geneve option for cases where pod or ns label was 0. But encountered various issues. We will look into that optimization post 5/30.